public/achievementInfo.php

```diff
@@ -85,13 +91,71 @@
 
 <body>
 <?php RenderHeader($userDetails); ?>
-<?php if ($permissions >= Permissions::Developer): ?>
+<?php if (($permissions >= Permissions::Developer || ($isSoleAuthor && $permissions >= Permissions::JuniorDeveloper && $achFlags === AchievementType::Unofficial))): ?>
     <script>
+        function updateAchievementDetails() {
+        var title = $('#titleinput').val();
+        var description = $('#descriptioninput').val();
+        var points = $('#pointsinput').val();
+        var validPoints = <?php echo json_encode(VALID_POINTS); ?>;
+        var titleLength = new Blob([title]).size;
+        var descLength = new Blob([descLength]).size;
+        var maxTitle = <?php echo MAX_TITLE; ?>;
+        var maxDescription = <?php echo MAX_DESCRIPTION; ?>;
+        showStatusMessage('Updating...');
+        if ( titleLength > maxTitle) {
+                showStatusFailure('Error: Title too long');
+                return;
+        } else if ( descLength > maxDescription) {
+                showStatusFailure('Error: Description too long');
+                return;
+        } else if ( points == '' || !validPoints.includes(Number(points))) {
+                showStatusFailure('Error: Invalid Points');
+                return;
+        }
+
+        $.ajax({
+          type: "POST",
+          url: '/request/achievement/update.php',
+          dataType: "json",
+          data: {
+            'a': <?= $achievementID ?>,
+            'f': 4,
+            'u': '<?= $user ?>',
+            'v': '',
+            'g': <?= $gameID ?>,
+            'l': '<?= $achFlags ?>',
+            'b': '<?= $badgeName ?>',
+            't': title,
+            'd': description,
+            'p': Number(points),
+            'm': '<?= $achMem ?>'
+          },
+          error: function (xhr, status, error) {
+            showStatusFailure('Error: ' + (error || 'unknown error'));
+          }
+        })
+          .done(function (data) {
+            if (!data.success) {
+              showStatusFailure('Error: ' + (data.error || 'unknown error'));
+              return;
+            }
+            window.location = window.location.href.split("?")[0] + "?e=modify_achievement_ok";
+          });
+        }
+
+    <?php if ($permissions >= Permissions::Developer): ?>
       function PostEmbedUpdate() {
         var url = $('#embedurlinput').val();
         url = replaceAll('http', '_http_', url);
 
         showStatusMessage('Updating...');
+
+        if ( url == '') {
+                showStatusFailure('Error: Blank url');
+                return;
```

Comment:
looks like this code should be using two space indents. probably the code added above too.